File: bruteforce/time.py
import time
import string

N = 26
letters = list(string.ascii_uppercase)

start = time.time()

count = 0
for i in range(N):
    for j in range(N):
        for k in range(N):
            for l in range(N):
                for m in range(N):
                    for n in range(N):
                        for o in range(N):
                            for p in range(N):
                                # print(f"{letters[i]}{letters[j]}{letters[k]}{letters[l]}{letters[m]}{letters[n]}{letters[o]}{letters[p]}")
                                count += 1

end = time.time()
time_taken = end - start

print(f"\nTotal kombinasi: {count}")
print(f"Waktu eksekusi: {time_taken:.6f} detik")
print(f"Kompleksitas waktu: O(n^8), dengan n = {N}")

File: bruteforce/time.c
#include <stdio.h>
#include <time.h>

#define N 26  // Jumlah huruf alfabet

int main() {
    char letters[N];
    for (int i = 0; i < N; i++) {
        letters[i] = 'A' + i;
    }

    clock_t start = clock();

    long long count = 0;
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < N; j++) {
            for (int k = 0; k < N; k++) {
                for (int l = 0; l < N; l++) {
                    for (int m = 0; m < N; m++) {
                        for (int n = 0; n < N; n++) {
                            for (int o = 0; o < N; o++) {
                                for (int p = 0; p < N; p++) {
                                    printf("%c%c%c%c%c%c%c%c\n", letters[i], letters[j], letters[k], letters[l], letters[m], letters[n], letters[o], letters[p]);
                                    count++;
                                }
                            }
                        }
                    }
                }
            }
        }
    }

    clock_t end = clock();
    double time_taken = (double)(end - start) / CLOCKS_PER_SEC;

    printf("\nTotal kombinasi: %lld\n", count);
    printf("Waktu eksekusi: %.6f detik\n", time_taken);
    printf("Kompleksitas waktu: O(n^8), dengan n = %d\n", N);

    return 0;
}
